search/query_parser.go

```diff
@@ -0,0 +1,90 @@
+package search
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// DID is not currently being used presently
+type DidHandle struct {
+	DID    string
+	Handle string
+}
+
+type SearchQuery struct {
+	FromUser    *DidHandle
+	QueryString string
+	Offset      int
+	Count       int
+}
+
+var /* const */ FromOperatorRegexp = regexp.MustCompile(`(?i)\bfrom:([\w\.]+)`)
+var /* const */ SearchQueryOffsetDefault = 0
+var /* const */ SearchQueryOffsetMin = 0
+var /* const */ SearchQueryCountDefault = 30
+var /* const */ SearchQueryCountMax = 100
+
+func paramsToPostsSearchQuery(queryString string, offsetParam string, countParam string) (*SearchQuery, error) {
```

Comment:
One stop shop for searching posts: parses out `from:` if present, ensure valid non-empty queries, handle offsets and counts.